File: Baekjoon/구현/1138.py
# https://www.acmicpc.net/problem/1138

n = int(input())
arr = list(map(int, input().split()))
dp = [0] * n

for i in range(n):
    for j in range(n):
        if arr[i] == 0 and dp[j] == 0:
            dp[j] = i + 1
            break
        elif dp[j] == 0:
            arr[i] -= 1
            
print(' '.join(map(str, dp)))

File: Baekjoon/구현/1138.cpp
// https://www.acmicpc.net/problem/1138

#include <iostream>
#include <vector>
using namespace std;

int main() {
    int n;
    cin >> n;

    vector<int> a(n);
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }

    vector<int> ans(n);
    for (int i = 0; i < n; i++) {
        int cnt = 0;
        for (int j = 0; j < n; j++) {
            if (cnt == a[i] && ans[j] == 0) {
                ans[j] = i + 1;
                break;
            }
            if (ans[j] == 0) {
                cnt++;
            }
        }
    }

    for (int i = 0; i < n; i++) {
        cout << ans[i] << ' ';
    }
    cout << '\n';

    return 0;
}